packages/framework/src/hooks/useTemplateData.ts

```diff
@@ -46,13 +48,25 @@ export const useTemplateData = ({
   }, [customScope, data, isDataString, context]);
   const rawData = useAtomValue(dataAtom);
   const namedData = useMemo(
-    () => map(rawData, (value: unknown) => ({ [name]: value })),
+    () => map(rawData, (val: unknown) => ({ [name]: val })),
     [name, rawData],
   );
+
   const evaluatedNamedData = useEvaluate({ namedData });
 
   return {
     rawData,
-    namedData: evaluatedNamedData.namedData,
+    namedData: map(evaluatedNamedData.namedData, (val) => ({
+      ...val,
+      value: value
```

Comment:
Pass this into useEvaluate. Make the key `_ensembleValue` to avoid conflicts with the evaluated keys.

Reply:
@evshi that will not work. because of, as per your suggestion,

```js
  const namedData = useMemo(
    () =>
      map(rawData, (val: unknown) => ({ [name]: val, _ensembleValue: value })),
    [name, rawData],
  );

  const evaluatedNamedData = useEvaluate({ namedData });
```

when I will pass `_ensembleValue` in useEvaluate like this way, where yaml is like,

```yaml
    Radio:
      id: radioGroup
      label: Radio Group With Template
      item-template:
        data: getData.body.users.slice(0, 4)
        name: user
        value: ${user.email}
        template:
          Row:
            children:
              - Image:
                  width: 25
                  height: 25
                  source: ${user.image}
              - Text:
                  text: ${user.firstName} ${user.lastName}
                  
```

so each object of namedData will become like this before getting evaluated by `useEvaluate`

```js
{
user: {...},
_ensembleValue: ${user.email}
}
```

then when we pass this to useEvaluate, it will extract expressions like this,
<img width="453" alt="image" src="https://github.com/user-attachments/assets/5747c753-f201-48e9-a452-ed03ce27ad43">

and then this expression will convert into bindings and will be try to evaluate with ensemble storage, ensemble env, defaultScreenContext, ensemble user, ensemble secrets, device and so on.

and eventually it return the `_ensembleValue` as `null` because this value is depend on `user` of `namedData` which is not available in context.